helix-term/src/ui/editor.rs

```diff
@@ -1413,13 +1415,31 @@ impl Component for EditorView {
                     let view = view_mut!(cx.editor, focus);
                     let doc = doc_mut!(cx.editor, &view.doc);
 
-                    view.ensure_cursor_in_view(doc, config.scrolloff);
-
                     // Store a history state if not in insert mode. This also takes care of
                     // committing changes when leaving insert mode.
                     if mode != Mode::Insert {
                         doc.append_changes_to_history(view.id);
                     }
+
+                    // If the current document has been changed, apply the changes to all views.
+                    // This ensures that selections in jumplists follow changes.
```

Comment:
Note: mouse clicks can also change view focus and are handled separately further below

Reply:
Ah ok, this will apply as soon as changes are added to history, I thought it was when focus is changed 👍🏻